### PR TITLE
feat: Revamp docs pages

### PR DESCRIPTION
**Pull request by gunnar2k**
This PR

* Drops `mdbook` and `swagger` in favor of `redocly` (https://redocly.com/docs/cli)  and migrates (some of) the markdown pages to be picked up by redocly when building the index.html.
* Drops `yamlfix` and `make format` since these kept messing up the newlines on the actual content of the pages.
* Moves some of custom mdbook markdown content into `docs/*.md` and picking them up from the OpenAPI spec
* Introduces new target in `Makefile`, `make preview` for simplifying developing the api docs page locally
* `make build` now builds the redocly page, creating a zero-dependency `public/index.html` 
* Updates some of the docs copy on some places, changing reference to `Vic.API` to `Vic.ai API`
* Reorganizes `Vendor tags` and `Vendor groups` into separate tabs underneat Vendors (see screenshots)
* Changes `General` tag to `Status`
* Changes name of tags to proper titles, rather than concatenated titles like `PurchaseOrders` (now `Purchase orders`) and `TrainingInvoices` (now `Invoices for training`)

## Screenshots

image

image

image

image

image

image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,23 +3,18 @@ all: build
...
 
-# Builds the entire static site documentation.
-build: validate
-	mdbook build
-	python scripts/openapi-build.py --input-file vic.api.v0.yaml --output-dir public --output-name vic.api.v0
+build:
+	npx --yes @redocly/cli build-docs vic.api.v0.yaml --output public/index.html
 
-format:
-	yamlfix vic.api.v0.yaml
```
Comment (warmwaffles) [inline Makefile:14]: Why are we dropping yamlfix? Does redocly do formatting?

File: Makefile [hunk 2]
```diff
@@ -3,23 +3,18 @@ all: build
...
 clean:
 	rm -rf public
-	rm -f src/vic.api.v0.html
 
 # Runs validations for the openapi specs
 validate:
-	yamlfix --check vic.api.v0.yaml
 	python -m openapi_spec_validator vic.api.v0.yaml
-	npx swagger-cli validate vic.api.v0.yaml
+	# npx --yes @redocly/cli lint vic.api.v0.yaml
```
Comment (warmwaffles) [inline Makefile:20]: Is the linter failing here?

Comment (warmwaffles) [reply Makefile:20]: Oh I see it is complaining.

**Review by warmwaffles — APPROVED** [review]

**Comment by warmwaffles** [comment]
@gunnar2k one thing that we need with redocly is to get the supplementary documentation working. I would like to provide more examples than what is generated by the openapi spec.

**Comment by gunnar2k** [comment]
@warmwaffles See how thats done in this PR with `main.md` referenced from the Open API spec

**Comment by gunnar2k** [comment]
Or do you mean we need to migrate more of the examples from the current supplementary pages?

**Comment by warmwaffles** [comment]
Some of the information for purchase orders that was delete could probably be included in the open api spec as a `ref:` to the md file. Redocly appears to take those markdown files an embed them into the spec which is nice.

**Comment by warmwaffles** [comment]
I pushed a change to start getting the redocly linter to pass. Some of these nullable fields are going to be challenging to fix because redocly does not like `allOf` and inheritance.

**Comment by warmwaffles** [comment]
@gunnar2k this now has the linter running.

**Comment by warmwaffles** [comment]
@gunnar2k I've re-added some of the supplemental documentation. This should be good to go now

**Pull request merged**